common/metrics/src/metric_server.rs

```diff
@@ -45,12 +45,12 @@ fn whitelist_metrics(
 // only return the whitelisted metrics
 fn whitelist_json_metrics(
     json_metrics: HashMap<String, String>,
-    whitelist: Vec<String>,
-) -> HashMap<String, String> {
-    let mut whitelist_metrics: HashMap<String, String> = HashMap::new();
-    for (key, val) in json_metrics.iter() {
-        if whitelist.contains(&key) {
-            whitelist_metrics.insert(key.to_string(), val.to_string());
+    whitelist: &'static [&'static str],
+) -> HashMap<&'static str, String> {
+    let mut whitelist_metrics: HashMap<&'static str, String> = HashMap::new();
+    for key in whitelist {
+        if let Some(metric) = json_metrics.get(*key) {
```

Comment:
I realize now that we could get away without doing another allocation by using the `remove` method on the HashMap instead of `get`...but in practice the number of these we're cloning will be small so I think this is ok to leave as is for now